tmt/steps/provision/localhost.py

```diff
@@ -1,7 +1,36 @@
-from click import echo
+import os
 
+from click import echo
 from tmt.steps.provision.base import ProvisionBase
+from tmt.utils import SpecificationError
+
 
 class ProvisionLocalhost(ProvisionBase):
+    """ Localhost provisioner """
+
+    def __init__(self, data, step):
+        super(ProvisionLocalhost, self).__init__(data, step)
+        self._prepare_map = {
+            'ansible': self._prepare_ansible,
+            'shell': self._prepare_shell,
+        }
+
     def execute(self, *args, **kwargs):
         self.run(self.join(args))
+
+    def _prepare_ansible(self, what):
+        """ Run ansible on localhost """
+        # note: we expect playbooks are placed relatively to the current directory
+        self.run(f'ansible-playbook -c local -i localhost, {what}', cwd=os.getcwd())
+
+    def _prepare_shell(self, what):
+        """ Run ansible on localhost """
+        # note: we expect playbooks are placed relatively to the current directory
+        self.run(what, cwd=os.getcwd())
```

Comment:
@psss isn't it like the `tmt` looks for git root and operates on it? But maybe I've misunderstood. If not, it is implemented wrong for ProvisionVagrant then, as it relies on the playbook being in `discover`.

Reply:
Yes, all paths should be relative to the metadata tree root. Will fix this.